workflows/hic.nf

```diff
@@ -0,0 +1,297 @@
+include { COMPARTMENTS } from '../subworkflows/local/compartments'
+include { TADS } from '../subworkflows/local/tads'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT NF-CORE MODULES/SUBWORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// MODULE: Installed directly from nf-core/modules
+//
+
+include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/modules/custom/dumpsoftwareversions/main'
+include { FASTQC  } from '../modules/nf-core/modules/fastqc/main'
+//include { MULTIQC } from '../modules/nf-core/modules/multiqc/main'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  CHANNELS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+Channel.fromPath( params.fasta )
+       .ifEmpty { exit 1, "Genome index: Fasta file not found: ${params.fasta}" }
+       .set { ch_fasta }
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    RUN MAIN WORKFLOW
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+// Info required for completion email and summary
+def multiqc_report = []
+
+workflow HIC {
+
+  ch_versions = Channel.empty()
+
+  //
+  // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+  //
+  INPUT_CHECK (
+    ch_input
+  )
+
+  INPUT_CHECK.out.reads.view()
+
+  //
+  // SUBWORKFLOW: Prepare genome annotation
+  //
+  PREPARE_GENOME(
+    ch_fasta,
+    ch_restriction_site
+  )
+  ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions)
+
+  //
+  // MODULE: Run FastQC
+  //
+  FASTQC (
+    INPUT_CHECK.out.reads
+  )
+  ch_versions = ch_versions.mix(FASTQC.out.versions)
+
+  //
+  // SUB-WORFLOW: HiC-Pro
+  //
+  HICPRO (
+    INPUT_CHECK.out.reads,
+    PREPARE_GENOME.out.index,
+    PREPARE_GENOME.out.res_frag,
+    PREPARE_GENOME.out.chromosome_size,
+    ch_ligation_site,
+    ch_map_res
+  )
+  ch_versions = ch_versions.mix(HICPRO.out.versions)
+
+  //
+  // SUB-WORKFLOW: COOLER
+  //
+  COOLER (
+    HICPRO.out.pairs,
+    PREPARE_GENOME.out.chromosome_size,
+    ch_map_res
+  )
+  ch_versions = ch_versions.mix(COOLER.out.versions)
+
+  //
+  // MODULE: HICEXPLORER/HIC_PLOT_DIST_VS_COUNTS
+  //
+  if (!params.skip_dist_decay){
+    COOLER.out.cool
+      .combine(ch_ddecay_res)
+      .view()
+      .filter{ it[0].resolution == it[2] }
+      .map { it -> [it[0], it[1]]}
+      .set{ ch_distdecay }
+
+    HIC_PLOT_DIST_VS_COUNTS(
+      ch_distdecay
+    )
+    ch_versions = ch_versions.mix(HIC_PLOT_DIST_VS_COUNTS.out.versions)
+  }
+
+  //
+  // SUB-WORKFLOW: COMPARTMENT CALLING
+  //
+  if (!params.skip_compartments){
+    COOLER.out.cool
+      .combine(ch_comp_res)
+      .filter{ it[0].resolution == it[2] }
+      .map { it -> [it[0], it[1], it[2]]}
+      .set{ ch_cool_compartments }
+
+    COMPARTMENTS(
```

Comment:
```suggestion
    COMPARTMENTS (
```
Not sure which style. In rnaseq and other's there's usually a space before the `(`, there were a few others of this before.